### PR TITLE
gcoap_fileserver: rename to nanocoap_fileserver

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

There isn't really anything GCoAP specific about the fileserver module, it can just as well be used with nanoCoAP. Move and rename it to reflect that.

### Testing procedure

Both `examples/gcoap_fileserver` and `examples/nanocoap_server` keep serving files, but the latter one without pulling in a GCoAP dependency.

### CODE REVIEW

File: sys/net/application_layer/nanocoap/fileserver.c [hunk 1]
```diff
@@ -143,10 +158,16 @@ static size_t gcoap_fileserver_error_handler(coap_pkt_t *pdu, uint8_t *buf, size
...
               code >> 5, code & 0x1f);
     }
-    return gcoap_response(pdu, buf, len, code);
+
+    if (_resp_init(pdu, buf, len, code)) {
+        return -1;
+    }
+    coap_opt_finish(pdu, COAP_OPT_FINISH_NONE);
+
+    return 0;
```
Comment (chrysn) [inline sys/net/application_layer/nanocoap/fileserver.c:170]: Is gcoap_response maybe just as due for a rename?

Comment (bergzand) [reply sys/net/application_layer/nanocoap/fileserver.c:170]: `gcoap_response` includes some observe handling which we would lose if we move the function to nanocoap. I'm not against adding a simple nanocoap_response in another PR though, as long as the difference with `gcoap` is documented somewhere.

**Review by Teufelchen1 — APPROVED** [review]
LGTM, tested on native manually (on my machine). However the `tests/net/gcoap_fileserver/` still fails. @MrKevinWeiss commented to me that this is a known issue, as seen in the makefile:
```
# This integration test uncovers a bug somwehere in our stack, but there
# are currently no resources to debug it, so blacklist it to keep CI running
TEST_ON_CI_BLACKLIST += all
```

**Comment by benpicco** [comment]
I made a mistake when moving away from `gcoap_response()` - [fixed](https://github.com/RIOT-OS/RIOT/compare/ece2a134b8290cb0ab1d2e23a8bc6be91f2fe5cb..c5ac57423692f5b301489713f28ee27398c4184e)

**Comment by Teufelchen1** [comment]
Yes! Works as intended. I think you should squash it.

**Pull request merged**